modules/packages/RangeChunk.chpl

```diff
@@ -0,0 +1,223 @@
+    if remPol == Pack {
+      chunkSize = nElems / nChunks;
+      if chunkSize * nChunks != nElems then
+        chunkSize += 1;
+    } else { // Mod
+      chunkSize = nElems / nChunks;
+      rem = nElems - chunkSize * nChunks;
+    }
+
+    for i in 0..#nChunks {
+      var chunk: 2*RT;
+      if remPol == Thru {
+        chunk = chunkOrderThru(nElems, nChunks, i);
+      } else if remPol == Pack {
+         chunk = chunkOrderPack(chunkSize, nElems, i);
+      } else { // Mod
+        chunk = chunkOrderMod(chunkSize, rem, nElems, nChunks, i);
+      }
+      yield chunk;
+    }
+  }
+
+  proc chunkOrder(
+    r: range(?RT, bounded, ?),
+    numChunks: integral,
+    idx: integral,
+    remPol: RemElems = Thru): 2*RT
+  {
+    if r.length == 0 || numChunks <= 0 || idx >= numChunks then
+      return (1: RT, 0: RT);
+
+    const nElems = r.length;
+    const nChunks = numChunks: RT;
+    const i = idx: RT;
+
+    if remPol == Thru {
+      return chunkOrderThru(nElems, nChunks, i);
+    } else if remPol == Pack {
+      var chunkSize = nElems / nChunks;
+      if chunkSize * nChunks != nElems then
+        chunkSize += 1;
+      return chunkOrderPack(chunkSize, nElems, i);
+    } else { // Mod
+      const chunkSize = nElems / nChunks;
+      const rem = nElems - chunkSize * nChunks;
+      return chunkOrderMod(chunkSize, rem, nElems, nChunks, i);
+    }
+  }
+
+  /*
+   * Query the chunk index of a value in the range.
+   * Returned value is zero-based.
+   */
+
+  proc whichChunk(
+    r: range(?RT, bounded, ?),
+    numChunks: integral,
+    val: integral,
+    remPol: RemElems = Thru): RT
+  {
+    assert(r.length != 0 && numChunks > 0);
+
+    const nElems = r.length;
+    const nChunks = numChunks: RT;
+    const i = r.indexOrder(val);
+
+    assert(i: int != -1);
+
+    if remPol == Thru {
+      return i * nChunks / nElems; 
+    } else if remPol == Pack {
+      var chunkSize = nElems / nChunks;
+      if chunkSize * nChunks != nElems then
+        chunkSize += 1;
+      return i / chunkSize;
+    } else { // Mod
+      const chunkSize = nElems / nChunks;
+      const chunkSizePlus = chunkSize + 1;
+      const rem = nElems - chunkSize * nChunks;
+      const splitPoint = rem * chunkSizePlus;
+      return if i < splitPoint
+        then i / chunkSizePlus 
+        else rem + (i - splitPoint) / chunkSize; 
+    }
+  }
+
+
+  /*
+   * Private helpers for order pairs and thereby ranges.
+   * Each corresponds with a remainder policy.
+   * i is a zero-based index.
+   */
+
+  // remainder elems distributed throughout 
+  private proc chunkOrderThru(
+    nElems: ?I,
+    nChunks: I,
+    i: I): (I, I)
+  {
+    const m = nElems * i;
+    const start = if i == 0
+      then 0: I
+      else ceilXDivByY(m, nChunks);
+    const end = if i == nChunks - 1
+      then nElems - 1
+      else ceilXDivByY(m + nElems, nChunks) - 1;
+    return (start, end);
+  }
+
+  private proc ceilXDivByY(x, y) return 1 + (x - 1)/y;
```

Comment:
There's a function that does this in the Math module called 'divceil'. You should be able to use it by default without 'use'ing any more modules.